onnxruntime/core/providers/xnnpack/nn/softmax.cc

```diff
@@ -30,13 +31,16 @@ bool IsQuantSoftmaxSupported(const NodeUnit& node_unit, const GraphViewer& graph
     // idealy, QlinearSoftmax or QDQSoftmax will keep this output scale and zp, but we have to handle some
     // qdq models converted from other framework
     auto [scale_tensor, zero_tensor] = GetQuantizationZeroPointAndScale(graph, node_unit.Outputs()[0]);
-    Initializer q_scale(*scale_tensor, node_unit.ModelPath());
-    if (fabs(*q_scale.data<float>() - 1.0f / 256.0f) > 0.0001f) {
+    std::optional<InitializerView> q_scale;
+    if (!InitializerView::Create(*scale_tensor, q_scale).IsOK()) {
+      break;
+    }
+    if (fabs(q_scale->DataAsSpan<float>()[0] - 1.0f / 256.0f) > 0.0001f) {
       break;
     }
-    if (scale_tensor) {
-      Initializer q_zp(*zero_tensor, node_unit.ModelPath());
-      if (*q_zp.raw_data() != 0) {
+    if (zero_tensor) {
+      std::optional<InitializerView> q_zp;
```

Comment:
>std::optional<InitializerView> q_zp; [](http://example.com/codeflow?start=6&length=36)

Just construct it on the stack. 
Can you use prepack mechanism to load this initializer?

Reply:
The use case for this is during GetCapability which happens at a time where initializers are TensorProto instances. PrePack happens during session state finalization which is way later and things have been converted to OrtValue.

Reply:
Offer two Ctors is better and we get rid of Optional<T>, we can SafeCreate(lazy initialized) without exception or just throw exception in the Ctor.